rego/rules/arm/monitor/key_vault_logging.rego

```diff
@@ -37,20 +37,42 @@ input_type = "arm"
 
 resource_type = "MULTIPLE"
 
-resources = fugue.resources("Microsoft.Insights/diagnosticSettings")
+key_vaults := fugue.resources("Microsoft.KeyVault/vaults")
+diagnostic_settings := fugue.resources("Microsoft.Insights/diagnosticSettings")
 
-is_invalid(resource) {
-	resource.TODO == "TODO" # FIXME
+retention_is_valid(retention) {
+	retention.enabled == true
+	retention.days >= 180
+}
+
+retention_is_valid(retention) {
+	retention.enabled == true
+	retention.days == 0
+}
+
+valid_key_vault_diagnostics := {id: ds |
+	ds := diagnostic_settings[id]
+	contains(lower(ds.scope), "microsoft.keyvault/vaults")
```

Comment:
Checking the scope this way and on line 65 is sub-optimal, but it should work with a variety of ways of defining it.